packages/blade-core/src/styles/BaseText/baseText.ts

```diff
@@ -0,0 +1,193 @@
+    },
+    letterSpacing: {
+      50: styles['letter-spacing-50'],
+      100: styles['letter-spacing-100'],
+    },
+    // Color is handled via compound variants - see getBaseTextClasses function
+    opacity: {
+      0: utilityClasses['opacity-0'],
+      25: utilityClasses['opacity-25'],
+      50: utilityClasses['opacity-50'],
+      75: utilityClasses['opacity-75'],
+      100: utilityClasses['opacity-100'],
+    },
+    numberOfLines: {
+      1: utilityClasses['line-clamp-1'],
+      2: utilityClasses['line-clamp-2'],
+      3: utilityClasses['line-clamp-3'],
+      4: utilityClasses['line-clamp-4'],
+      5: utilityClasses['line-clamp-5'],
+      6: utilityClasses['line-clamp-6'],
+      7: utilityClasses['line-clamp-7'],
+      8: utilityClasses['line-clamp-8'],
+      9: utilityClasses['line-clamp-9'],
+      10: utilityClasses['line-clamp-10'],
+    },
+  },
+  defaultVariants: {
+    fontSize: 200,
+    lineHeight: 100,
+    fontWeight: 'regular',
+    fontFamily: 'text',
+    fontStyle: 'normal',
+    textDecorationLine: 'none',
+  },
+});
+
+/**
+ * Convert color token to CSS class name (scoped via CSS module)
+ * Example: 'surface.text.gray.normal' -> styles['color-surface-text-gray-normal']
+ */
+function colorTokenToClassName(color: string): string | undefined {
+  if (color === 'currentColor') {
+    return styles['color-current'];
+  }
+  // Convert dot notation to kebab-case class name
+  const withHyphens = color.replace(/\./g, '-');
+  const className = `color-${kebabCase(withHyphens)}`;
+  // Access the scoped class name from CSS module
+  // The CSS module is imported in this file, so styles object contains scoped class names
+  return styles[className];
+}
+
+/**
+ * Generate all classes for BaseText component
+ * This is the single source of truth for all BaseText styling
+ * Everything is class-based - no data attributes or inline styles
+ */
+export function getBaseTextClasses(props: BaseTextVariants & { className?: string }): string {
+  const { className, color, opacity, numberOfLines, ...cvaProps } = props;
+
+  // Generate CVA classes
+  const cvaVariants = {
+    fontSize: cvaProps.fontSize,
+    lineHeight: cvaProps.lineHeight,
+    fontWeight: cvaProps.fontWeight,
+    fontFamily: cvaProps.fontFamily,
+    fontStyle: cvaProps.fontStyle,
+    textDecorationLine: cvaProps.textDecorationLine,
+    textAlign: cvaProps.textAlign,
+    textTransform: cvaProps.textTransform,
+    wordBreak: cvaProps.wordBreak,
+    letterSpacing: cvaProps.letterSpacing,
+    opacity:
+      opacity !== undefined ? (Math.round(opacity * 100) as 0 | 25 | 50 | 75 | 100) : undefined,
+    numberOfLines:
+      numberOfLines !== undefined && numberOfLines >= 1 && numberOfLines <= 10
+        ? (numberOfLines as 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10)
+        : undefined,
+  };
+
+  const classes = [
+    baseTextStyles(cvaVariants),
+    // Generate color class name dynamically
+    color ? colorTokenToClassName(color) : undefined,
```

Comment:
is there no way to still keep color in CVA without having to define each prop? like is there any way to do

```js
{
  color: (color) => colorTokenToClassName(color)
}
```